core/server/master/src/main/java/alluxio/master/backup/BackupWorkerRole.java

```diff
@@ -240,6 +240,10 @@ private CompletableFuture<Void> handleRequestMessage(BackupRequestMessage reques
         LOG.info(
             "Initiating catching up of journals to consistent sequences before starting backup. {}",
             requestMsg.getJournalSequences());
+        // TODO(jiacheng): If the journal is corrupted,
+        //  the catchup attempt will throw ExecutionException wrapping an RuntimeException
+        //  That ex will be caught and sent back to primary, but
+        //  the standby master will NOT be crashed and that's a problem
         CatchupFuture catchupFuture = mJournalSystem.catchup(requestMsg.getJournalSequences());
         CompletableFuture.runAsync(catchupFuture::waitTermination)
```

Comment:
Here if I don't crash the UfsJournalCheckpointThread and if an uncaught exception/RuntimeException is thrown from the CatchupThread, the exception will be caught and sent back to the master and the standby master will keep running. I'm not sure if there could be a problem because I can't tell what the uncaught exception can be.

Reply:
My current thought is, if the backup fails because the journal catchup failed due to something like a corrupted journal, then the master should crash (but maybe not crashed by logic here but from the catchup thread). If the backup fails due to IOException then maybe we want the master to not crash.